cli/azd/internal/repository/infra_confirm.go

```diff
@@ -194,3 +200,26 @@ func (i *Initializer) infraSpecFromDetect(
 
 	return spec, nil
 }
+
+func (i *Initializer) detectPortInDockerfile(
+	filePath string) []int {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return []int{}
+	}
+	defer file.Close()
+
+	var result []int
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "EXPOSE") {
+			var port int
+			_, err := fmt.Sscanf(line, "EXPOSE %d", &port)
```

Comment:
This looks great!

For completeness, I'd make sure to handle:
- Multiple ports
- Transmission protocol (udp/tcp) being present

See [EXPOSE](https://docs.docker.com/reference/dockerfile/#expose) for reference.